On POST, replaced ownerId with lookup of the common name from OpenAM.

kshakir opened the pull request:
DSDEV-1819

Config files are more properties like, using = instead of mixed in json.

kshakir committed:
On POST, replaced ownerId with lookup of the common name from OpenAM. DSDEV-1819
Config files are more properties like, using = instead of mixed in json.

kshakir committed:
Updated build to allow passing some java properties to sbt-revolver.
Also ignoring the .sbtopts file.

jmthibault79 commented on src/test/scala/org/broadinstitute/dsde/vault/common/openam/OpenAMClientSpec.scala:
```diff
@@ -0,0 +1,111 @@
+        auth.get.successUrl shouldNot be(empty)
+      }
+
+      "Authentication with a bad host name should fail" in {
```
Why are the first 3 params tested but not the last 3?

kshakir replied:
I imagined configuration wise, outside of our for apollo, those are optional. From reading the various hits on google regarding OpenAM:
- Sub-realm seemed to be optional
- Authentication via custom indexes was optional too

I'm happy to put more Vault (or Broad?) specific checks here, but if it moved to a real common, not sure if future setups will have values for those. Definitely open to suggestions, as I really don't know too much about OpenAM.

jmthibault79 replied:
OK, I'm fine with existing tests.

jmthibault79 commented on src/main/scala/org/broadinstitute/dsde/vault/common/openam/OpenAMClient.scala:
```diff
@@ -0,0 +1,70 @@
+    waitResponse(pipeline(Post(uri, Map.empty[String, String])))
+  }
+
+  def lookupUsernameCN(deploymentUri: String, token: String, id: String, realm: Option[String]) = {
```
It took me some time to figure out CN = "Common Name".  Could you add a short comment?  Perhaps also at UsernameCNResponse?

jmthibault79 commented on src/test/scala/org/broadinstitute/dsde/vault/datamanagement/services/AnalysisServiceSpec.scala:
```diff
@@ -58,8 +59,8 @@ class AnalysisServiceSpec extends DataManagementDatabaseFreeSpec with AnalysisSe
-          Post(pathBase, Analysis(badInput, metadata, files)) ~> sealRoute(ingestRoute) ~> check {
-            status === BadRequest
+        Post(pathBase, Analysis(badInput, metadata, files)) ~> OpenAMSession ~> sealRoute(ingestRoute) ~> check {
+          status === BadRequest
```
When should we use `===` vs. `should be()` ?

kshakir replied:
Reading up on this now myself. Open to suggestions: http://www.scalatest.org/user_guide/using_matchers#checkingEqualityWithMatchers

kshakir replied:
Hmm... well there goes my confidence with ScalaTest / spray-testkit.

All passing:

``` scala
status === BadRequest
status === NotFound
status === Success
```

Fail:

``` scala
status should be(BadRequest)
```

Pass:

``` scala
status should be(InternalServerError)
```

jmthibault79 replied:
Interesting: none of the examples use bare `===`, only `should ===`.  If `should` is required, we need to avoid the bare version.

jmthibault79 commented on src/test/scala/org/broadinstitute/dsde/vault/datamanagement/services/UnmappedBAMServiceSpec.scala:
```diff
@@ -17,7 +18,7 @@ class UnmappedBAMServiceSpec extends DataManagementDatabaseFreeSpec with Unmappe
       var createdId: Option[String] = None
```
The line above this has `ownerId` metadata.  I'd prefer replacing it with some other arbitrary metadata, in case it's hiding any residual `ownerId` issues.

jmthibault79 commented on src/main/scala/org/broadinstitute/dsde/vault/common/openam/OpenAMDirectives.scala:
```diff
@@ -0,0 +1,37 @@
+    }
+  }
+
+  val commonNameByCookie = tokenFromCookie map {
```
Naming: I would prefer commonNameFromCookie

jmthibault79 commented on src/main/scala/org/broadinstitute/dsde/vault/common/openam/OpenAMDirectives.scala:
```diff
@@ -0,0 +1,37 @@
+    _.content
+  }
+
+  val optionalTokenFromCookie = optionalCookie(OpenAMConfig.tokenCookie) map {
```
Naming: tokenFromOptionalCookie would be more correct here, if I understand what it's doing.

jmthibault79 commented on src/main/scala/org/broadinstitute/dsde/vault/common/openam/OpenAMDirectives.scala:
```diff
@@ -0,0 +1,37 @@
+    case token => commonNameFromToken(token)
+  }
+
+  val optionalCommonNameByCookie = optionalTokenFromCookie map {
```
Naming: commonNameFromOptionalCookie ?  Assuming that is what it's doing.

kshakir committed:
Cleanup of commit replacing ownerId with OpenAM calls.
Tests no longer using "===" (with missing should), now using "should be".
Renamed directives under OpenAMDirectives.
Added brief doc line regarding common name (CN).
Removed all traces of ownerId.

kshakir commented:
Fixed up the above, and added a commit to make it easier to quickly test with local `.sbtopts` files.

Comments welcome. If all is well, including https://github.com/broadinstitute/cmi-chef-repo/pull/65, then merge away on both. Or let me know if one would prefer a single squash and I can merge them.

jmthibault79 commented:
Looks good.  OK to merge after Chef is done.
